### DISCUSSION TITLE
Add support for new video modes

### DISCUSSION THREAD

**Discussion by viti95** [viti95/FastDoom#123] [Ideas]
VGA Planar Mode 13h should be faster than Mode-Y in some cases, for example Rendition Verite cards doesn't like this mode at all. Same happens for Cyrix MediaGX SOCs. Maybe it's also possible add support for VESA modes.

- [x] Text 80x25 (double height, 16 colors, CGA, EGA, VGA)
- [x] Text 80x50 (16 colors, VGA)
- [x] CGA 320x200 4 colors
- [x] EGA 320x200 16 colors
- [x] CGA 640x200 B&W
- [x] Hercules 640x400 B&W
- [x] VGA Planar Mode 13h
- [x] VESA (2.0, 320x200 Banked & LFB)
- [x] MDA 80x25 (double height, 4 monochrome levels)
- [ ] MDA 80x25 (double height, 8 colors, only for some IBM MDA cards)
- [x] ATI Small Wonder, 640x200, 16 colors, dithered
- [x] EGA 640x200 16 colors, dithered
- [ ] Add support for 64 colors on EGA cards
- [ ] IBM 8514/A (Wireframe?, Automap?)
- [ ] IBM Professional Graphics Controller (Maybe only usable for the automap)
- [ ] IBM XGA / XGA-2
- [ ] Yamaha v6366 (CGA compatible card with 320x200 and 256 color support)
- [ ] IBM CGA Composite 256 color mode (80x100, based on "512" color mode)

**Comment by viti95** [comment]
Reopen for new video modes

**Comment by viti95** [comment]
Reopen again for new video modes

**Comment by neuralnetworkz** [comment]
Maybe some of these cga black magic graphic modes?

Crtc - 8088 MPH
https://www.pouet.net/prod.php?which=65371


Crtc - area5150
https://www.pouet.net/prod.php?which=91938

Some info how it's done:

http://www.reenigne.org/blog/1k-colours-on-cga-how-its-done/

**Comment by neuralnetworkz** [comment]
Also accelerated IBM xga/xga-2 maybe?
I have an IBM ps2/e with xga-2 1Mb to test on

**Comment by viti95** [comment]
Yep, XGA is another candidate to have an accelerated version of Doom. [@LuigiThirty](https://twitter.com/LuigiThirty) is doing it right now, using the Doom base code instead of FastDoom. Maybe I can add support later, as I don't have any XGA hardware available.

**Comment by neuralnetworkz** [comment]
Sounds great 😁

**Comment by neuralnetworkz** [comment]
there's also this yamaha 6366 CGA chip which supposedly can display 320x200 in 256 Colours. I have one (but no monitor yet, just a portable PC lcd panel connected)
It can do Plantronics mode for a start. 
[V6366_Yamaha.pdf](https://github.com/viti95/FastDoom/files/9505327/V6366_Yamaha.pdf)
![IMG_5640](https://user-images.githubusercontent.com/6433825/188863823-0a1add96-e51d-438d-8a8e-679e457d9fc0.jpeg)

**Comment by viti95** [comment]
That's interesting, I'll add it to the to-do list, but will be hard to implement, since there are no emulators compatible with this videocard, nor do I have a v6366 ISA card.
